src/renderer/object/instanced_model.rs

```diff
@@ -49,53 +52,173 @@ impl<M: ForwardMaterial> InstancedModel<M> {
             instance_buffer1: InstanceBuffer::new(context)?,
             instance_buffer2: InstanceBuffer::new(context)?,
             instance_buffer3: InstanceBuffer::new(context)?,
+            instance_buffer4: InstanceBuffer::new(context)?,
             aabb,
             aabb_local: aabb.clone(),
             transformation: Mat4::identity(),
             transformations: transformations.to_vec(),
+            subtexture: None,
+            subtextures: None,
             material,
         };
-        model.update_transformations(transformations);
-        model.update_aabb();
+        model.set_transformations(transformations);
         Ok(model)
     }
 
     ///
-    /// Updates the transformations applied to each model instance before they are rendered.
-    /// The model is rendered in as many instances as there are transformation matrices.
+    /// Use [set_transformations()](InstancedModel::set_transformations()) instead.
     ///
+    #[deprecated]
     pub fn update_transformations(&mut self, transformations: &[Mat4]) {
+        self.set_transformations(&transformations);
+    }
+
+    ///
+    /// Store the transformations applied to each model instance before they are rendered.
+    /// The model is rendered in as many instances as there are transformation matrices.
+    ///
+    pub fn set_transformations(&mut self, transformations: &[Mat4]) {
         self.transformations = transformations.to_vec();
-        self.instance_count = transformations.len() as u32;
+        self.update_buffers();
+    }
+
+    ///
+    /// Store the transformations applied to each model instance before they are rendered.
+    /// The model is rendered in as many instances as there are transformation matrices.
+    /// Must (eventually) call [update_buffers()](InstancedModel::update_buffers()) afterward.
+    ///
+    pub fn add_transformation(&mut self, transformation: &Mat4) {
```

Comment:
I would rather have an `add_instance` function that takes a transformation and an optional texture transform.

Reply:
This way new transforms can be added without needing to add a redundant texture transform. All instances can still use the same texture, or a single instance can use many texture transforms.

Reply:
Donezo